draft-ietf-rats-corim.md

```diff
@@ -1186,6 +1186,78 @@ Evidence.
 {::include cddl/domain-membership-triple-record.cddl}
 ~~~
 
+
+#### Group Membership Triple {#sec-comid-triple-group-membership}
+
+In order to model hierarchical device composition, CoRIM authors need to identify all components in the composite device (see Composite Device in {{-rats-arch}}). A group describes all target environments that a certain attesting environment is generating evidence about. Consequently, all target environments in question are associated with a common domain via the group-membership-triple.
+
+~~~ cddl
+group-membership-triple-record = [
+  relative-top-layer-root: environment-map
+  relative-sub-nodes: [ + environment-map ]
+]
+~~~
+
+~~~
+      absolute root
+          env1
+
+  env2    env3
+
+  env4
+
+  =>
+
+group-membership-triple-record = [
+  relative-top-layer-root: env1
+  relative-sub-tree-nodes: [ env2, env3 ]
+]
+
+group-membership-triple-record = [
+  relative-top-layer-root: env2
+  relative-sub-tree-nodes: [ env4 ]
+]
+~~~
+
+In this example above, env1 is the Attesting Environment for env2 and env3. Consequently, env1 cannot generate valid Evidence about env4.
+After being appraised as trustworthy by the Verifier, env2 can act as the Attesting Environment for env4 and Evidence generated by env2 about env4 can be appraised by the Verifier.
+
+The environment-map that is  the relative-top-layer-root functions as the group name. Hence, it is expressed as an environment that can appear as one of the `sub-tree` elements of other `group-membership-triple-record`, thus allowing recursive composition.
+
+The scope of a single group-membership-triple-record encompassed exactly two adjacent layers in an Attester.
+
+#### Multi-Environment Conditional (MEC) Endorsements Triple {#sec-comid-triple-mec-endorsements}
+
+The semantics of the Multi-Environment Conditional (MEC) Endorsements Triple is as follows:
+
+> "IF accepted state matches the `cond` value, THEN the `env` is associated with the endorsed value `val`."
+
+~~~ cddl
+multi-env-conditional-endorsement-triple-record = [
+  env: environment-map                    ; the environment to which the endorsed value
+                                          ; applies
+  val: measurement-values-map             ; the endorsed value
+  cond: [ + stateful-environment-record ] ; all target environments along with
+                                          ; their expected state that need to match
+]
```

Comment:
I have a version of this which adds multiple conditional endorsements (with the same condition for each). I think that is likely to be widely used, and for the cost of 1 additional byte is probably worthwhile.

Reply:
added multiple cardinality to `val`